src/main/java/org/embulk/input/google_ads/GoogleAdsReporter.java

```diff
@@ -282,14 +288,92 @@ public List<String> buildWhereClauseConditions(PluginTask task, Map<String, Stri
     }
 
     public void connect()
+    {
+        this.client = buildClient(task.getLoginCustomerId().orElseGet(() -> getLoginCustomerId(task.getCustomerId())));
+    }
+
+    private Long getLoginCustomerId(String customerId)
+    {
+        try (CustomerServiceClient client = buildClient(null).getLatestVersion().createCustomerServiceClient()) {
+            return client.listAccessibleCustomers(ListAccessibleCustomersRequest.newBuilder().build())
+                    .getResourceNamesList()
+                    .stream()
+                    .map(CustomerName::parse)
+                    .map(CustomerName::getCustomerId)
+                    .map(this::getLoginCustomerClients)
+                    .flatMap(Collection::stream)
+                    .collect(Collectors.groupingBy(LoginCustomerClient::getCustomerClientId, Collectors.maxBy(Comparator.comparing(LoginCustomerClient::getLevel))))
```

Comment:
If duplicate clients are found, select the client with the maximum distance between manager account and client.